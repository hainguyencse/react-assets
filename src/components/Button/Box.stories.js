import React from 'react';
import { action } from '@storybook/addon-actions';
import Button from '.';

export default [
  {
    title: 'Default',
    component: <Button onClick={action('clicked')}>Default</Button>
  },
  {
    title: 'Primary',
    component: <Button onClick={action('clicked')} type="primary">Primary</Button>
  },
  {
    title: 'Danger',
    component: <Button onClick={action('clicked')} type="danger">Danger</Button>
  },
];
