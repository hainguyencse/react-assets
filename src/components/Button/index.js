import React from 'react';
import PropTypes from 'prop-types';

const Button = ({ children, onClick, block, type, disabled }) => (
  <button
    type="button" className={`btn ${block ? 'btn-block' : ''} btn-${type} ${disabled ? 'disabled' : ''}`}
    onClick={disabled ? () => {} : onClick }>
    {children}
  </button>
);

Button.propTypes = {
  children: PropTypes.node,
  onClick: PropTypes.func,
  block: PropTypes.bool,
  disabled: PropTypes.bool,
  type: PropTypes.oneOf(['default', 'primary', 'warning', 'danger']),
};

Button.defaultProps = {
  children: 'Button',
  onClick: () => {},
  block: false,
  disabled: false,
  type: 'default',
};

export default Button;
