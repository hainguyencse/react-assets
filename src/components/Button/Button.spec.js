import React from 'react';
import Button from './index';

describe('Button', () => {
  test('should render properly', () => {
    // eslint-disable-next-line
    const wrapper = shallow(<Button type="primary" block>Click me</Button>);

    // eslint-disable-next-line
    expect(wrapper).toMatchSnapshot();
  });
});
