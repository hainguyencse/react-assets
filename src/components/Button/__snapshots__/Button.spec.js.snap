// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button should render properly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    block={true}
    displayType="default"
    size="nor"
    type="primary"
  >
    Click me
  </Button>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Click me",
      "className": "btn btn-block btn-default  btn-nor",
      "type": "primary",
    },
    "ref": null,
    "rendered": "Click me",
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Click me",
        "className": "btn btn-block btn-default  btn-nor",
        "type": "primary",
      },
      "ref": null,
      "rendered": "Click me",
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
